src/flb_upstream.c

```diff
@@ -237,6 +237,39 @@ struct flb_upstream *flb_upstream_create(struct flb_config *config,
     return u;
 }
 
+/*
+ * Checks whehter a destinate URL should be proxied.
+ */
+int flb_should_proxy_for_host(const char *host, const char *proxy, const char *no_proxy)
+{
+    /* No HTTP_PROXY, should not set up proxy for the upstream `host`. */
+    if (proxy == NULL) {
+        return FLB_FALSE;
+    }
+
+    /* No NO_PROXY with HTTP_PROXY set, should set up proxy for the upstream `host`. */
+    if (no_proxy == NULL) {
+        return FLB_TRUE;
+    }
+
+    /* NO_PROXY=`*`, it matches all hosts. */
+    if (strcmp(no_proxy, "*") == 0) {
+        return FLB_FALSE;
+    }
+
+    /* check the URL list in the NO_PROXY  */
+    char *no_proxy_url = strtok(no_proxy, ",");
```

Comment:
- variable needs to be declared on top

- do not use strtok, instead go with flb_slist_split_tokens() API

Reply:
DONE. Thanks for letting me know, I used `flb_slist_split_string` for this now and also enhanced its unit tests.